metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -100,9 +102,48 @@ final class RenameProvider(
   ): Future[WorkspaceEdit] = {
     val source = params.getTextDocument.getUri.toAbsolutePath
     compilations.compilationFinished(source).flatMap { _ =>
-      definitionProvider
+      val defininionFuture = definitionProvider
         .definition(source, params, token)
-        .flatMap { definition =>
+      defininionFuture.flatMap { definition =>
+        if (definition.symbol.isLocal) {
+
+          // We have to check if local symbol doesn't override any non-local symbols
+          // If it does, we can't use PC for local rename
+          val overridenSymbols =
```

Comment:
This is used for anonymous classes, like in this tests 
https://github.com/scalameta/metals/blob/81aaecb7173bc50edb9394889d86f61472f6aeaa/tests/unit/src/test/scala/tests/RenameLspSuite.scala#L446-L464
Method inside anonymous class in local, but it overrides nonlocal one and we want to rename all of them. We can't use standard PC rename, since it won't rename global symbols or even other local symbols. If this local symbol overrides something nonlocal, we use `RenameProvider.rename` on it and it should rename everything, both local and not (thats why we can't yet remove handling local symbols in rename provider). If all overridden symbols are local, we use PC rename on each of them.